source/simulator/melt.cc

```diff
@@ -1363,6 +1752,11 @@ namespace aspect
     if (SimulatorAccess<dim> *sim = dynamic_cast<SimulatorAccess<dim>*>(boundary_fluid_pressure.get()))
       sim->initialize_simulator (simulator_object);
     boundary_fluid_pressure->initialize ();
+
+    if (use_discontinuous_p_c)
```

Comment:
Try moving this to `parse_parameters` if it does not work we need to look at core.cc (simulator constructor)